autoload/d2.vim

```diff
@@ -30,7 +30,7 @@ endfunction
 
 function! d2#init_block_string_syntaxes() abort
   let b:included_syntaxes = get(b:, 'included_syntaxes', [])
-  if type(g:d2_block_string_syntaxes) != v:t_dict && empty(g:d2_block_string_syntaxes)
+  if exists('g:d2_block_string_syntaxes') == 0 || type(g:d2_block_string_syntaxes) != v:t_dict || empty(g:d2_block_string_syntaxes) == 1
```

Comment:
I reckon this is called from `syntax/d2.vim`, wouldn't it be better to call `d2#add_default_block_string_syntaxes` just before `d2#init_block_string_syntaxes` gets called from `syntax/d2.vim`?

Reply:
`plugin/d2.vim` is only called once, but `syntax/d2.vim` will be loaded for each open `d2` file.

Reply:
Ah right I missed that sorry.

I guess if there was a way for `d2#add_default_block_string_syntaxes` to be called before the syntax file is loaded, it would be the ideal fix.

Right now, I think the issue stems from the fact that the syntax file is loaded before the `plugin/d2.vim` file is loaded.

In any case, don't mind me, this is safe to merge imo and fixes #3 , thanks!